pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/IntermediateResultsBlock.java

```diff
@@ -45,7 +50,7 @@
  * The <code>IntermediateResultsBlock</code> class is the holder of the server side inter-segment results.
  */
 public class IntermediateResultsBlock implements Block {
-  private DataSchema _selectionDataSchema;
+  private DataSchema _dataSchema;
```

Comment:
For aggregation, you don't really need data schema. Data schema is used to passing the data type for each column. For aggregation, the data type can be derived from the aggregation function.
For the column order, you can refer to the `SelectionOperatorUtils` where order-by expressions are always put in front of other expressions. With that you don't even need to maintain the column index.

Reply:
I prefer not to make the change about reordering columns in this iteration. Can we take it up in following PRs? I've already noted it down in the list of possible optimizations.
The data schema is mainly needed for the Table structure.

Reply:
As per offline discussion, added TODO